src/main/scala/org/bitcoins/core/crypto/TransactionSignatureComponent.scala

```diff
@@ -23,14 +24,14 @@ trait TransactionSignatureComponent {
  *
    * @return
    */
-  def inputIndex : Int
+  def inputIndex : UInt32
 
   /**
    * The script signature being checked
  *
    * @return
    */
-  def scriptSignature = transaction.inputs(inputIndex).scriptSignature
+  def scriptSignature = transaction.inputs(inputIndex.underlying.toInt).scriptSignature
```

Comment:
does the .toInt need to be here? Theoretically this could cause an overflow if we have > `2^31` inputs. Ideally it should be .toLong